stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/ui/common/feature/channel/info/ChannelInfoController.kt

```diff
@@ -0,0 +1,420 @@
+                    _events.tryEmit(
+                        ChannelInfoEvent.HideChannelError(message = error.message),
+                    )
+                }
+        }
+    }
+
+    /**
+     * Unhides the channel.
+     */
+    public fun unhideChannel() {
+        logger.d { "[unhideChannel]" }
+        scope.launch {
+            channelClient.show().await()
+                .onError { error ->
+                    logger.e { "[unhideChannel] error: ${error.message}" }
+                    _events.tryEmit(
+                        ChannelInfoEvent.UnhideChannelError(message = error.message),
+                    )
+                }
+        }
+    }
+
+    /**
+     * Leaves the channel with an optional quit message.
+     *
+     * @param quitMessage The system message to send when leaving the channel.
+     */
+    public fun leaveChannel(quitMessage: Message?) {
+        logger.d { "[leaveChannel] quitMessage: ${quitMessage?.text}" }
+
+        val onError: (Error) -> Unit = { error ->
+            logger.e { "[leaveChannel] error: ${error.message}" }
+            _events.tryEmit(
+                ChannelInfoEvent.LeaveChannelError(message = error.message),
+            )
+        }
+
+        requireCapability(
+            permission = { canLeaveChannel },
+            onError = onError,
+        ) {
+            scope.launch {
+                runCatching {
+                    requireNotNull(chatClient.getCurrentOrStoredUserId()) { "User not connected" }
+                }.onSuccess { currentUserId ->
+                    channelClient.removeMembers(
+                        memberIds = listOf(currentUserId),
+                        systemMessage = quitMessage,
+                    ).await()
+                        .onSuccess { _events.tryEmit(ChannelInfoEvent.LeaveChannelSuccess) }
+                        .onError(onError)
+                }.onFailure { cause ->
+                    onError(Error.ThrowableError(message = cause.message.orEmpty(), cause = cause))
+                }
+            }
+        }
+    }
+
+    /**
+     * Deletes the channel.
+     */
+    public fun deleteChannel() {
+        logger.d { "[deleteChannel]" }
+
+        val onError: (Error) -> Unit = { error ->
+            logger.e { "[deleteChannel] error: ${error.message}" }
+            _events.tryEmit(
+                ChannelInfoEvent.DeleteChannelError(message = error.message),
+            )
+        }
+
+        requireCapability(
+            permission = { canDeleteChannel },
+            onError = onError,
+        ) {
+            scope.launch {
+                channelClient.delete().await()
+                    .onSuccess { _events.tryEmit(ChannelInfoEvent.DeleteChannelSuccess) }
+                    .onError(onError)
+            }
+        }
+    }
+
+    private fun List<Member>.filterNotCurrentUser() =
+        filter { member -> member.user.id != chatClient.getCurrentOrStoredUserId() }
```

Comment:
`getCurrentOrStoredUserId()` is mainly used to obtain an userId even if the ChatClient is in a "not-ready" state.
We shouldn't abuse to use this method. I would prefer to not use this method at all

Reply:
The goal was to make it unit-testable. The current `ChatClient` dependencies are really hard to unit-test because they are too coupled and requires a lot of integration.
Unit testing a class which depends on `ChatClient` should be easier.
I would love to hear your suggestions...

Reply:
I will use `getCurrentUser` instead 👍🏻